DQMOffline/Trigger/plugins/DiJetMonitor.cc

```diff
@@ -0,0 +1,294 @@
+
+  histname = "jetptTag"; histtitle = "Tag Jet Pt";
+  bookME(ibooker,jetptTagME_,histname,histtitle,dijetpt_binning_.nbins,dijetpt_binning_.xmin, dijetpt_binning_.xmax);
+  setMETitle(jetptTagME_,"Pt_tag [GeV]","events ");
+
+  histname = "jetptPrb"; histtitle = "Probe Jet Pt";
+  bookME(ibooker,jetptPrbME_,histname,histtitle,dijetpt_binning_.nbins,dijetpt_binning_.xmin,dijetpt_binning_.xmax);
+  setMETitle(jetptPrbME_,"Pt_prb [GeV]","events");
+
+  histname = "jetptAsym"; histtitle = "Jet Pt Asymetry";
+  bookME(ibooker,jetptAsyME_,histname,histtitle,asy_binning_.nbins,asy_binning_.xmin, asy_binning_.xmax);
+  setMETitle(jetptAsyME_,"(pt_prb - pt_tag)/(pt_prb + pt_tag)","events");
+
+  histname = "jetetaPrb"; histtitle = "Probe Jet eta";
+  bookME(ibooker,jetetaPrbME_,histname,histtitle,dijet_eta_binning_.nbins,dijet_eta_binning_.xmin, dijet_eta_binning_.xmax);
+  setMETitle(jetetaPrbME_,"Eta_probe #eta","events");
+
+  histname = "jetetaTag"; histtitle = "Tag Jet eta";
+  bookME(ibooker,jetetaTagME_,histname,histtitle,dijet_eta_binning_.nbins,dijet_eta_binning_.xmin, dijet_eta_binning_.xmax);
+  setMETitle(jetetaTagME_,"Eta_tag #eta","events");
+
+  histname = "ptAsymVSetaPrb"; histtitle = "Pt_Asym vs eta_prb";
+  bookME(ibooker,jetAsyEtaME_,histname,histtitle,asy_binning_.nbins, asy_binning_.xmin, asy_binning_.xmax, dijet_eta_binning_.nbins, dijet_eta_binning_.xmin,dijet_eta_binning_.xmax);
+  setMETitle(jetAsyEtaME_,"(pt_prb - pt_tag)/(pt_prb + pt_tag)","Eta_probe #eta");
+
+  histname = "etaPrbVSphiPrb"; histtitle = "eta_prb vs phi_prb";
+  bookME(ibooker,jetEtaPhiME_,histname,histtitle,dijet_eta_binning_.nbins, dijet_eta_binning_.xmin, dijet_eta_binning_.xmax, dijet_phi_binning_.nbins, dijet_phi_binning_.xmin,dijet_phi_binning_.xmax);
+  setMETitle(jetEtaPhiME_,"Eta_probe #eta","Phi_probe #phi");
+
+  histname = "jetphiPrb"; histtitle = "Probe Jet phi";
+  bookME(ibooker,jetphiPrbME_,histname,histtitle,dijet_phi_binning_.nbins,dijet_phi_binning_.xmin, dijet_phi_binning_.xmax);
+  setMETitle(jetphiPrbME_,"Phi_probe #phi","events");
+
+  // Initialize the GenericTriggerEventFlag
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "DataFormats/Math/interface/deltaR.h" // For Delta R
+void DiJetMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+ 
+  v_jetpt.clear();
```

Comment:
hi @bobaekim - please make these variables local variables (or just remove them completely and fill pt_1 (etc) directly in your loop